### ISSUE TITLE
Allow plugin ordering

### ISSUE THREAD

**Issue by mtulio** [vmware-tanzu/sonobuoy#1684]
**Describe the solution you'd like**

> References #631 #432 #39

Plugins should be able to run in a predetermined order.
 
**Anything else you would like to add:**
 
As well described on #631 , the plugins could be able to run in a certain order configured on their manifests. The default behavior could stay the current execution (all in parallel), but it would be nice to run, for example in serial mode or a mix: A->B->C->(D&E)->F.

Also could be interesting if there's one flag if one plugin has failed, the following ones will not be executed (maybe also marked as failed?). It can be useful in preflights plugins which usually run on the begging to make some prechecks (like minimal resources, minimal performance checks, etc).

**Comment by johnSchnake** [comment]
Thanks for the submission and for the specific ideas related to use cases and it is a good point to raise about whether or not to continue with execution or to stop if one fails.

Like we talked about in slack, this idea always leads me to think about passing data along the chain as well but that has to be out of scope for now if we want to get this in.

**Comment by johnSchnake** [comment]
Here was my PR for this in the past; I might take a look again. I believe it did work but it looks like I was partially concerned about implementing a feature like this without fully thinking about all the potential ramifications. We never got more push for it from the community so it just fizzled out after that (which is why I'm glad you reached out about it).

**Comment by johnSchnake** [comment]
Oop, didnt realize I forgot to paste the link: https://github.com/vmware-tanzu/sonobuoy/pull/627

**Comment by johnSchnake** [comment]
@andrewyunt has picked this up and is tweaking my old PR to start. I think we'll have a PR soon to iterate on/discuss.
